{
	"id": "ALAS2022-2023-274",
	"advisories": [
		{
			"id": "ALAS2022-2023-274",
			"title": "Amazon Linux 2022 - ALAS2022-2023-274: Important priority package update for python3.10",
			"description": "Package updates are available for Amazon Linux 2022 that fix the following vulnerabilities:\nCVE-2022-45061:\n\tAn issue was discovered in Python before 3.11.1. An unnecessary quadratic algorithm exists in one path when processing some inputs to the IDNA (RFC 3490) decoder, such that a crafted, unreasonably long name being presented to the decoder could lead to a CPU denial of service. Hostnames are often supplied by remote servers that could be controlled by a malicious actor; in such a scenario, they could trigger excessive CPU consumption on the client attempting to make use of an attacker-supplied supposed hostname. For example, the attack payload could be placed in the Location header of an HTTP response with status code 302. A fix is planned in 3.11.1, 3.10.9, 3.9.16, 3.8.16, and 3.7.16.\n\nCVE-2022-42919:\n\tPython 3.9.x before 3.9.16 and 3.10.x before 3.10.9 on Linux allows local privilege escalation in a non-default configuration. The Python multiprocessing library, when used with the forkserver start method on Linux, allows pickles to be deserialized from any user in the same machine local network namespace, which in many system configurations means any user on the same machine. Pickles can execute arbitrary code. Thus, this allows for local user privilege escalation to the user that any forkserver process is running as. Setting multiprocessing.util.abstract_sockets_supported to False is a workaround. The forkserver start method for multiprocessing is not the default start method. This issue is Linux specific because only Linux supports abstract namespace sockets. CPython before 3.9 does not make use of Linux abstract namespace sockets by default. Support for users manually specifying an abstract namespace socket was added as a bugfix in 3.7.8 and 3.8.3, but users would need to make specific uncommon API calls in order to do that in CPython before 3.9.\n",
			"severity": [
				{
					"type": "vendor",
					"source": "linux-security@amazon.com",
					"vendor": "Important"
				}
			],
			"references": [
				{
					"source": "linux-security@amazon.com",
					"url": "https://alas.aws.amazon.com/AL2022/ALAS-2023-274.html"
				},
				{
					"source": "linux-security@amazon.com",
					"url": "http://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-42919"
				},
				{
					"source": "linux-security@amazon.com",
					"url": "http://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-45061"
				}
			]
		}
	],
	"vulnerabilities": [
		{
			"id": "CVE-2022-42919",
			"references": [
				{
					"source": "linux-security@amazon.com",
					"url": "http://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-42919"
				}
			]
		},
		{
			"id": "CVE-2022-45061",
			"references": [
				{
					"source": "linux-security@amazon.com",
					"url": "http://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-45061"
				}
			]
		}
	],
	"detection": [
		{
			"ecosystem": "amazon:2022",
			"vulnerable": true,
			"package": {
				"name": "python3.10-debugsource",
				"repositories": [
					"amazonlinux"
				],
				"architectures": [
					"aarch64"
				]
			},
			"affected": {
				"type": "rpm",
				"range": [
					{
						"lt": "0:3.10.9-1.amzn2022.0.1"
					}
				],
				"fixed": [
					"0:3.10.9-1.amzn2022.0.1"
				]
			}
		},
		{
			"ecosystem": "amazon:2022",
			"vulnerable": true,
			"package": {
				"name": "python3.10-debug",
				"repositories": [
					"amazonlinux"
				],
				"architectures": [
					"aarch64"
				]
			},
			"affected": {
				"type": "rpm",
				"range": [
					{
						"lt": "0:3.10.9-1.amzn2022.0.1"
					}
				],
				"fixed": [
					"0:3.10.9-1.amzn2022.0.1"
				]
			}
		},
		{
			"ecosystem": "amazon:2022",
			"vulnerable": true,
			"package": {
				"name": "python3.10-debuginfo",
				"repositories": [
					"amazonlinux"
				],
				"architectures": [
					"aarch64"
				]
			},
			"affected": {
				"type": "rpm",
				"range": [
					{
						"lt": "0:3.10.9-1.amzn2022.0.1"
					}
				],
				"fixed": [
					"0:3.10.9-1.amzn2022.0.1"
				]
			}
		},
		{
			"ecosystem": "amazon:2022",
			"vulnerable": true,
			"package": {
				"name": "python3.10-libs",
				"repositories": [
					"amazonlinux"
				],
				"architectures": [
					"aarch64"
				]
			},
			"affected": {
				"type": "rpm",
				"range": [
					{
						"lt": "0:3.10.9-1.amzn2022.0.1"
					}
				],
				"fixed": [
					"0:3.10.9-1.amzn2022.0.1"
				]
			}
		},
		{
			"ecosystem": "amazon:2022",
			"vulnerable": true,
			"package": {
				"name": "python3.10",
				"repositories": [
					"amazonlinux"
				],
				"architectures": [
					"aarch64"
				]
			},
			"affected": {
				"type": "rpm",
				"range": [
					{
						"lt": "0:3.10.9-1.amzn2022.0.1"
					}
				],
				"fixed": [
					"0:3.10.9-1.amzn2022.0.1"
				]
			}
		},
		{
			"ecosystem": "amazon:2022",
			"vulnerable": true,
			"package": {
				"name": "python3.10-tkinter",
				"repositories": [
					"amazonlinux"
				],
				"architectures": [
					"aarch64"
				]
			},
			"affected": {
				"type": "rpm",
				"range": [
					{
						"lt": "0:3.10.9-1.amzn2022.0.1"
					}
				],
				"fixed": [
					"0:3.10.9-1.amzn2022.0.1"
				]
			}
		},
		{
			"ecosystem": "amazon:2022",
			"vulnerable": true,
			"package": {
				"name": "python3.10-devel",
				"repositories": [
					"amazonlinux"
				],
				"architectures": [
					"aarch64"
				]
			},
			"affected": {
				"type": "rpm",
				"range": [
					{
						"lt": "0:3.10.9-1.amzn2022.0.1"
					}
				],
				"fixed": [
					"0:3.10.9-1.amzn2022.0.1"
				]
			}
		},
		{
			"ecosystem": "amazon:2022",
			"vulnerable": true,
			"package": {
				"name": "python3.10-idle",
				"repositories": [
					"amazonlinux"
				],
				"architectures": [
					"aarch64"
				]
			},
			"affected": {
				"type": "rpm",
				"range": [
					{
						"lt": "0:3.10.9-1.amzn2022.0.1"
					}
				],
				"fixed": [
					"0:3.10.9-1.amzn2022.0.1"
				]
			}
		},
		{
			"ecosystem": "amazon:2022",
			"vulnerable": true,
			"package": {
				"name": "python3.10-test",
				"repositories": [
					"amazonlinux"
				],
				"architectures": [
					"aarch64"
				]
			},
			"affected": {
				"type": "rpm",
				"range": [
					{
						"lt": "0:3.10.9-1.amzn2022.0.1"
					}
				],
				"fixed": [
					"0:3.10.9-1.amzn2022.0.1"
				]
			}
		}
	],
	"data_source": "amazon"
}
